pkgs/development/embedded/srec/default.nix

```
@@ -0,0 +1,23 @@
+{ lib, clangStdenv, fetchFromGitHub }:
+clangStdenv.mkDerivation rec {
+  name = "srec-${version}";
+  version = "2015-03-06";
+
+  src = fetchFromGitHub {
+    owner = "arkku";
+    repo = "srec";
+    rev = "4bf276fd8abc2b3e57ef0e0b7c44dcacd8d810ee";
+    sha256 = "HnQ64CPOA7m4FkMIRI3bx8kySY6uUihVKlfKYqxufSc=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp bin2srec $out/bin
+  '';
+  meta = with lib; {
+    description = "Binaries to convert Motorola SREC (S-Record) hex files";
+    homepage = "https://github.com/arkku/srec";
+    license = licenses.mit;
+    maintainers = with maintainers; [ simoneruffini ];
```

```suggestion

  meta = {
    description = "Binaries to convert Motorola SREC (S-Record) hex files";
    homepage = "https://github.com/arkku/srec";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ simoneruffini ];
```
Using `with lib;` with `meta` is discouraged.

Please set the [`mainProgram`](https://nixos.org/manual/nixpkgs/unstable/#var-meta-mainProgram) and [`platforms`](https://nixos.org/manual/nixpkgs/unstable/#var-meta-platforms)

---

It doesn't have a main program it has two complementary programs, one to convert from binary file to srec format and the other in the inverse direction